fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/navigation/RoadPositionFactory.java

```diff
@@ -72,36 +64,32 @@ public static IRoadPosition createAlongRoute(
         Validate.notNull(cnc, "The CentralNavigationComponent must not be null");
         Validate.notNull(currentRoute, "The route must not be null");
         Validate.notNull(currentPosition, "The currentPosition must not be null");
-        Validate.notNull(currentPosition.getPreviousNode(), "The currentPosition must provide the previously passed node");
+        Validate.notNull(currentPosition.getConnectionId(), "The currentPosition must provide the id of the connection");
         Validate.isTrue(distance > 0d, "The distance must be greater than 0");
 
-        String nodeId;
-        String from = null;
-        String to = null;
         double untilStop = 0 - currentPosition.getOffset() - distance;
-        for (Iterator<String> nodeIdIterator = currentRoute.getNodeIdList().iterator(); nodeIdIterator.hasNext(); ) {
-            nodeId = nodeIdIterator.next();
-            if (from != null) {
-                to = nodeId;
-                untilStop += cnc.getPositionOfNode(from).distanceTo(cnc.getPositionOfNode(to));
+
+        String connectionToStopOn = null;
+        for (String connectionId : currentRoute.getConnectionIds()) {
+            if (connectionId.equals(currentPosition.getConnectionId())) {
+                connectionToStopOn = connectionId;
+            }
+
+            if (connectionToStopOn != null) {
```

Comment:
This could be handled in the above if-statement to reduce code complexity.

Reply:
Nope, this would break the algorithm:

Suppose a route of connections (`a` (10m),`b` (40m),`c` (30m),`d` (10m)) and the vehicle is currently on connection `b`. Now imagine you want to get the connection which is 50m away from the current position. The loop will jump into the first `if` as soon as it finds `b`. Then it goes into second if and reduces `untilStop`. Now we need another step in the iteration (`c`). The first if is `false` and it still goes into the second if and finds that `untilStop` is greater than 50 and returns `c`.

Reply:
Ahhh I didn't look at `untilStop`. I didn't see that it is initialized with a negative number.
Nvm my first comment then. :)